### PR TITLE
Attempt `noarch_platforms` build

### LINKED ISSUE #13: [BUG?] Versions >=0.4.0 cannot be installed using only conda-forge channels
### Comment:

Version `>=0.4.0` is noarch, and depends on tensorflow `>=2.5.0`. 

However, [only version 1.14.0 of tensorflow](https://anaconda.org/conda-forge/tensorflow) is available via conda-forge for Windows OSs (this appears to conflict with the [claims of the tensorflow-feedstock badges though](https://github.com/conda-forge/tensorflow-feedstock), but conda cannot locate any versions on Windows above 1.14.0).

This means that any packages to be uploaded to conda-forge that require cellfinder-core >=0.4.0 [have unresolvable dependencies on Windows](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=732367&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=357).

**Workarounds**
For conda-forge recipes: forcing Windows to use cellfinder-core<0.4.0 will bypass the issue but leave the user with a broken package, unless they follow the step below to install a compatible version of tensorflow first. This also makes any recipes illegible for noarch.

A compatible version of tensorflow can be installed via pip on Windows, so it is possible to install cellfinder-core>=0.4.0 manually by:
- `pip install tensorflow`
- `conda install cellfinder-core>=0.4.0`

But of course, conda-forge recipes cannot use non-conda channels, so this is not available when uploading new recipes to `staged-recipes`.

### PR DESCRIPTION
**Pull request by willGraham01**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

## Do not merge until `cellfinder-core` version is updated (see [this checklist](https://github.com/conda-forge/cellfinder-core-feedstock/issues/13#issuecomment-1621371663))

- [x] Waiting on https://github.com/brainglobe/cellfinder-core/pull/191
- [ ] Waiting on PyPI upload of `cellfinder-core` with https://github.com/brainglobe/cellfinder-core/pull/186 changes

Once ready:
- Closes #13

[As suggested](https://github.com/conda-forge/cellfinder-core-feedstock/issues/13#issuecomment-1620634236) in #13, we can try to make use of `noarch_platforms` builds to ship a functional package on as many OSes as possible.

Currently only Windows has the `tensorflow` incompatibility issue, `unix` operating systems appear to be able to fetch `tensorflow` just fine. So we can specify a "unix-only" dependency of `tensorflow`, which _should_ allow us to ship a working package on those platforms.

Windows installs will retain a broken package though. Also all the final checks can only pass on `unix` for the same reason, so they've been scoped to `unix` only.

### CODE REVIEW

**Comment by willGraham01** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,18 +31,21 @@ requirements:
...
 test:
   imports:
-    - cellfinder_core
+    - cellfinder_core  # [unix]
   commands:
     - pip list
-    - pip check
-    - cellfinder_download --help
-    - cellfinder_train --help
+    - pip check  # [unix]
```
Comment (adamltyson) [inline recipe/meta.yaml:46]: Is it worth adding some checks for the other packages that will work across OSs here? E.g.:
```
napari --info
brainglobe --help
brainreg --help
```

**Review by adamltyson — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (willGraham01) [reply recipe/meta.yaml:46]: For the meta-package & `cellfinder-napari`/`cellfinder` I agree (at least for `cellfinder-core` dependencies we expect to be installed regardless).

As for here, `pip list` is still included in the checks so we can confirm everything that should be fetched on Windows is being fetched.

Comment (adamltyson) [reply recipe/meta.yaml:46]: haha ignore me, I didn't know which feedstock I was reviewing! Feel free to merge as is.

**Comment by willGraham01** [comment]
@adamltyson can you hit merge? I don't have write access to this repo as I'm not a maintainer

**Pull request merged**